web/src/app/schedules/temp-sched/TempSchedAddShiftsStep.tsx

```diff
@@ -126,52 +126,29 @@ export default function TempSchedAddShiftsStep({
   // that makes the network request.
   function fieldErrors(s = submitted): FieldError[] {
     const result: FieldError[] = []
-
-    if (!shift) {
-      return result
+    const requiredMsg = 'this field is required'
+    const add = (field: string, message: string): void => {
+      result.push({ field, message } as FieldError)
     }
 
+    if (!shift) return result
     if (s) {
-      const message = 'this field is required'
-      if (!shift.userID) {
-        result.push({
-          field: 'userID',
-          message,
-        } as FieldError)
-      }
-      if (!shift.start) {
-        result.push({
-          field: 'start',
-          message,
-        } as FieldError)
-      }
-      if (!shift.end) {
-        result.push({
-          field: 'end',
-          message,
-        } as FieldError)
-      }
-
-      return result
+      if (!shift.userID) add('userID', requiredMsg)
+      if (!shift.start) add('start', requiredMsg)
+      if (!shift.end) add('end', requiredMsg)
     }
 
-    if (!isISOAfter(shift.end, shift?.start)) {
-      result.push({
-        field: 'end',
-        message: 'must be after shift start time',
-      } as FieldError)
+    if (!isISOAfter(shift.end, shift.start)) {
+      add('end', 'must be after shift start time')
+    }
+    if (!isISOBefore(shift.start, shift.end)) {
+      add('start', 'must be before shift end time')
```

Comment:
this is redundant, do a single check

Reply:
I'm not sure I agree with this change, as I think we should only display the error on the field they are typing in, rather than both.

Reply:
I'm not sure I agree with this change, as I would expect only the field I am typing in to error, rather than display on both.